code/campaignclassic/src/main/java/com/adobe/marketing/mobile/ManualCarouselTemplateNotificationBuilder.java

```diff
@@ -0,0 +1,531 @@
+                        context,
+                        0,
+                        clickIntent,
+                        PendingIntent.FLAG_MUTABLE | PendingIntent.FLAG_UPDATE_CURRENT);
+
+        clickIntent.setAction(CampaignPushConstants.IntentActions.MANUAL_CAROUSEL_RIGHT_CLICKED);
+        final PendingIntent pendingIntentRightButton =
+                PendingIntent.getBroadcast(
+                        context,
+                        0,
+                        clickIntent,
+                        PendingIntent.FLAG_MUTABLE | PendingIntent.FLAG_UPDATE_CURRENT);
+
+        expandedLayout.setOnClickPendingIntent(R.id.leftImageButton, pendingIntentLeftButton);
+        expandedLayout.setOnClickPendingIntent(R.id.rightImageButton, pendingIntentRightButton);
+
+        NotificationCompat.Builder builder;
+
+        if (handlingIntent) {
+            // we need to create a silent notification as this will be re-displaying a notification
+            // rather than showing a new one.
+            // the silent sound is set on the notification channel and notification builder.
+            Log.trace(CampaignPushConstants.LOG_TAG, SELF_TAG, "Building a silent notification.");
+            builder =
+                    new NotificationCompat.Builder(
+                            context,
+                            CampaignPushConstants.DefaultValues.SILENT_NOTIFICATION_CHANNEL_ID);
+
+            AEPPushNotificationBuilder.setSound(context, builder, customSound, true);
+        } else {
+            builder = new NotificationCompat.Builder(context, channelId);
+            AEPPushNotificationBuilder.setSound(context, builder, customSound, false);
+        }
+
+        builder.setNumber(badgeCount)
+                .setAutoCancel(true)
+                .setStyle(new NotificationCompat.DecoratedCustomViewStyle())
+                .setCustomContentView(smallLayout)
+                .setCustomBigContentView(expandedLayout);
+
+        AEPPushNotificationBuilder.setSmallIcon(
+                context,
+                builder,
+                smallIcon,
+                smallIconColor); // Small Icon must be present, otherwise the notification will not
+        // be displayed.
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            AEPPushNotificationBuilder.setVisibility(builder, visibility);
+        }
+
+        // if API level is below 26 (prior to notification channels) then notification priority is
+        // set on the notification builder
+        if (android.os.Build.VERSION.SDK_INT < android.os.Build.VERSION_CODES.O) {
+            builder.setPriority(NotificationCompat.PRIORITY_HIGH)
+                    .setVibrate(
+                            new long[0]); // hack to enable heads up notifications as a HUD style
+            // notification requires a tone or vibration
+        }
+
+        return builder;
+    }
+
+    private static Intent createClickIntent(
```

Comment:
Do you foresee this method being invoked by callers other than `createNotificationBuilder` ? If not, can you we move the click intent creating logic into `createNotificationBuilder` itself?

Reply:
sure, i'll move it within